content/en/blog/single-or-monolith-serverless-functions /index.md

```diff
@@ -0,0 +1,135 @@
+With our lives surrounded by apps, the need for quality software is growing.
+Afterall, better the code, better the app. Microservices and serverless are helping make applications more scalable and faster along with making their deployments easier.
+
+However, there’s one question that most developers have today is whether to use single or monolith serverless functions in their applications.
+In this post, I will try to help answer that.
+
+## The Basics
+
+There are a variety of software design principles that enable developers to write better code. One of the popular principles is **SOLID**.
+It’s a mnemonic acronym for: *Single Responsibility Principle,Open Closed Principle, Liskov Substitution Principle , Interface Segregation Principle and Dependency Inversion Principle.*
+
+Design principles like these are not limited to any specific domain or technology.
+Whether you are writing a low level code that would be etched on a chip or a serverless function that would be running on the cloud.
+The decision of applying any of these principles depend on the developer and the use case, however following them will only make the code better and optimized.
+
+While all of them are equally important, that one that is of interest to us in this blog post is the single responsibility principle - *each function must have only one specific operation that ir performs*
+This infact forms the basis of microservices and serverless.
+
+With serverless functions becoming mainstream today, many developers have a question whether to use single or monolith serverless functions for their application? Before we dive into the answer, let’s understand what we mean by single and monolith serverless functions.  
+
+{{< figure src="/images/featured/single-monolith-featured.png" alt="Single or Monolith Serverless Functions - What should you choose?" height="600" width="1000">}}
+
+## Single vs Monolith Serverless Functions
+
+A **single serverless function is responsible for only one function in your application**. For example and for simplicity, let us assume that you are developing a calculator.
+In that case, you will have four different functions - *add, subtract, multiply and divide* - which would be 4 different serverless functions with individual files. So you’ll have a main function that would interact with the user the other functions based on the input.
+
+In case of a **monolith serverless function approach, there is just one serverless function that has all the sub functions in it**.
+There’s a logic within the function to call the other sub functions based on the request.
+In this case you’ll have a single serverless function with add, subtract, divide and multiply functions.
+The client facing application will call this function which will return the output after processing.
+
+### Sample application
+
+To understand the single and monolith serverless functions, we will build a simple Stock Keeper application developed using Python using Fission functions.
```

Comment:
Should we call it `inventory status application` ?

Stock I felt for a moment, stock market stock.